pragma-oracle/src/oracle/oracle.cairo

```diff
@@ -514,18 +522,59 @@ mod Oracle {
         }
 
         // @notice aggregate all the entries for a given data type, with a given aggregation mode
+        // @notice The aggregation mode `ConversionRate` is set for tokenized vault, with pricing made based on STRK price and associated conversion rate fetched 
+        // @notice onchain. 
         // @param data_type: an enum of DataType (e.g : DataType::SpotEntry(ASSET_ID) or DataType::FutureEntry((ASSSET_ID, expiration_timestamp)))
         // @param aggregation_mode: the aggregation method to be used (e.g. AggregationMode::Median(()))
         // @returns a PragmaPricesResponse, a structure providing the main information for an asset (see entry/structs for details)
         fn get_data(
             self: @ContractState, data_type: DataType, aggregation_mode: AggregationMode
         ) -> PragmaPricesResponse {
-            let sources = IOracleABI::get_all_sources(self, data_type);
-            let prices_response: PragmaPricesResponse = IOracleABI::get_data_for_sources(
-                self, data_type, aggregation_mode, sources
-            );
+            if aggregation_mode == AggregationMode::ConversionRate {
+                // Query median for STRK/USD
+                let sources = IOracleABI::get_all_sources(self, DataType::SpotEntry('STRK/USD'));
+                let response: PragmaPricesResponse = IOracleABI::get_data_for_sources(
+                    self, DataType::SpotEntry('STRK/USD'), AggregationMode::Median(()), sources
+                );
 
-            prices_response
+                //  Extract base asset
+                let asset: felt252 = match data_type {
+                    DataType::SpotEntry(asset) => asset,
+                    DataType::FutureEntry(_) => panic_with_felt252('Set only for Spot entries'),
+                    DataType::GenericEntry(_) => panic_with_felt252('Set only for Spot entries'),
+                };
+
+                // Get base currency and pool
+                let base_asset: felt252 = self.get_pair(asset).base_currency_id;
+                assert(base_asset != 0, 'Asset not registered');
+                let pool_address: ContractAddress = self.tokenized_vault.read((base_asset, 'STRK'));
+                assert(
+                    pool_address != starknet::contract_address_const::<0>(),
+                    'No pool address for given token'
+                );
+                let pool = IERC4626Dispatcher { contract_address: pool_address };
+
+                // Compute adjusted price
+                // `preview_mint` takes as argument an e18 and returns an e18
+                // We operate under u256 to avoid overflow
+                let price: u256 = response.price.into()
+                    * pool.preview_mint(1000000000000000000)
```

Comment:
what's the argument of preview mint ?

Reply:
it's 1  share, expressed with 18 decimals